src/OrbitBase/Logging.cpp

```diff
@@ -5,15 +5,65 @@
 #include "OrbitBase/Logging.h"
 
 #include <array>
+#include <system_error>
 
 #include "absl/base/const_init.h"
 #include "absl/debugging/stacktrace.h"
 #include "absl/debugging/symbolize.h"
+#include "absl/strings/str_cat.h"
 #include "absl/strings/str_format.h"
 #include "absl/synchronization/mutex.h"
 
 static absl::Mutex log_file_mutex(absl::kConstInit);
 std::ofstream log_file;
+const int kLogFileLifetimeMinutes = 60 * 24 * 7;  // one week
+
+std::optional<int> GetLogFileLifetimeMinutes(std::string log_file_name) {
+  unsigned first_delimiter_pos = log_file_name.find_first_of(kLogFileNameDelimiter);
+  unsigned last_delimiter_pos = log_file_name.find_last_of(kLogFileNameDelimiter);
+  std::string tm_str =
+      log_file_name.substr(first_delimiter_pos + 1, last_delimiter_pos - first_delimiter_pos - 1);
+  if (tm_str.empty()) return std::nullopt;
+
+  std::istringstream istr(tm_str);
+  int year, month, day, hour, minute, second;
+  char delimiter;
+  // The timestamp in the log file name follows a format of "%Y_%m_%d_%H_%M_%S".
+  if (istr >> year >> delimiter >> month >> delimiter >> day >> delimiter >> hour >> delimiter >>
+      minute >> delimiter >> second) {
```

Comment:
Use early return

if (! ... ) return std::nullopt;

...
In that case there is one less nested block.

Reply:
Thanks for the suggestion! Changed to use `absl::FormatTime` and `absl::ParseTime` to simpliy the code here.